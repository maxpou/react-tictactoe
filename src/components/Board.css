.board-row:after {
  clear: both;
  content: "";
  display: table;
}

.status {
  margin-bottom: 10px;
}

.kbd-navigation .square:focus {
  background: #ddd;
}
