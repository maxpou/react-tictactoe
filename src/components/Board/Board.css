.board {
  min-width: 100px;
}

.board-row:after {
  clear: both;
  content: "";
  display: table;
}
