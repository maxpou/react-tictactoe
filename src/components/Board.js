import React from 'react'
import getWinner from '../services/get-winner'
import Square from './Square'
import './Board.css'

export default class Board extends React.Component {
  constructor () {
    super()
    this.init()
  }

  init () {
    this.state = {
      squares: Array(9).fill(null),
      xIsNext: true
    }
  }

  renderSquare (i) {
    return (
      <Square
        value={this.state.squares[i]}
        onClick={() => this.handleClick(i)}
      />
    )
  }

  handleClick (i) {
    const squares = this.state.squares.slice()

    if (getWinner(squares) || squares[i]) {
      return
    }
    squares[i] = this.state.xIsNext ? 'X' : 'O'
    this.setState({
      squares: squares,
      xIsNext: !this.state.xIsNext
    })
  }

  reset () {
    console.log('reset')
    this.init()
    console.log(this.state)
  }

  render () {
    const winner = getWinner(this.state.squares)
    let status
    if (winner) {
      status = 'Winner: ' + winner
    } else {
      status = 'Next player: ' + (this.state.xIsNext ? 'X' : 'O')
    }

    return (
      <div>
        <div className='status'>{status}</div>
        <div className='board-row'>
          {this.renderSquare(0)}
          {this.renderSquare(1)}
          {this.renderSquare(2)}
        </div>
        <div className='board-row'>
          {this.renderSquare(3)}
          {this.renderSquare(4)}
          {this.renderSquare(5)}
        </div>
        <div className='board-row'>
          {this.renderSquare(6)}
          {this.renderSquare(7)}
          {this.renderSquare(8)}
        </div>
        <div className='restart'>
          <button onClick={() => this.reset()}>Restart</button>
        </div>
      </div>
    )
  }
}
