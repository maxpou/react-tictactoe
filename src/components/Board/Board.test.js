import React from 'react'
import { shallow, mount } from 'enzyme'
import Board from './Board'

describe('Board.js', () => {
  it('renders without crashing', () => {
    const squares = Array(9).fill(null)
    const wrapper = shallow(<Board squares={squares} />)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should call a function on click', () => {
    const squares = Array(9).fill(null)
    const onClickEvent = jest.fn()
    const wrapper = mount(<Board squares={squares} onClick={onClickEvent} />)
    const button = wrapper.find('button.square').first()
    button.simulate('click')

    expect(onClickEvent).toBeCalledWith(0)
  })
})
