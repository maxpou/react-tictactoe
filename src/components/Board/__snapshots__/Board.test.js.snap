// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board.js renders without crashing 1`] = `"<div class=\\"board\\"><div class=\\"board-row\\"><button class=\\"square\\"></button><button class=\\"square\\"></button><button class=\\"square\\"></button></div><div class=\\"board-row\\"><button class=\\"square\\"></button><button class=\\"square\\"></button><button class=\\"square\\"></button></div><div class=\\"board-row\\"><button class=\\"square\\"></button><button class=\\"square\\"></button><button class=\\"square\\"></button></div></div>"`;
